### PR TITLE
Migrated the environment variable MLFLOW_DEPLOYMENT_FLAVOR_NAME

### PR DESCRIPTION
**Pull request by Sandeep1005**
## Related Issues/PRs

 #9315

## What changes are proposed in this pull request?

Migrated the environment variable MLFLOW_DEPLOYMENT_FLAVOR_NAME from mlflow/mlflow/models/container/__init__.py to mlflow/environment_variables.py

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by Sandeep1005** [comment]
@harupy please review this and let me know if I need to change anything.

File: mlflow/models/container/__init__.py [hunk 1]
```diff
@@ -70,8 +69,8 @@ def _serve(env_manager):
     model_config_path = os.path.join(MODEL_PATH, MLMODEL_FILE_NAME)
     m = Model.load(model_config_path)
 
-    if DEPLOYMENT_CONFIG_KEY_FLAVOR_NAME in os.environ:
-        serving_flavor = os.environ[DEPLOYMENT_CONFIG_KEY_FLAVOR_NAME]
+    if MLFLOW_DEPLOYMENT_FLAVOR_NAME.defined():
+        serving_flavor = MLFLOW_DEPLOYMENT_FLAVOR_NAME.get()
     else:
         # Older versions of mlflow may not specify a deployment configuration
         serving_flavor = pyfunc.FLAVOR_NAME
```
Comment (harupy) [inline mlflow/models/container/__init__.py:76]:
```suggestion
    serving_flavor = MLFLOW_DEPLOYMENT_FLAVOR_NAME.get() or pyfunc.FLAVOR_NAME
```

File: mlflow/environment_variables.py [hunk 2]
```diff
@@ -398,3 +398,9 @@ def get(self):
 MLFLOW_RECIPES_EXECUTION_TARGET_STEP_NAME = _EnvironmentVariable(
     "MLFLOW_RECIPES_EXECUTION_TARGET_STEP_NAME", str, None
 )
+
+#: Spevifies the deployment flavour??
+#: (default ``None``)
+MLFLOW_DEPLOYMENT_FLAVOR_NAME = _EnvironmentVariable(
+    "MLFLOW_DEPLOYMENT_FLAVOR_NAME", str, None
+)
```
Comment (harupy) [inline mlflow/environment_variables.py:406]:
```suggestion
#: Specifies the flavor name to deploy.
#: (default ``None``)
MLFLOW_DEPLOYMENT_FLAVOR_NAME = _EnvironmentVariable(
    "MLFLOW_DEPLOYMENT_FLAVOR_NAME", str, None
)

```

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**